dsc_lib/src/dscresources/resource_manifest.rs

```diff
@@ -50,6 +50,9 @@ pub struct ResourceManifest {
 
 #[derive(Debug, Clone, PartialEq, Deserialize, Serialize, JsonSchema)]
 pub enum InputKind {
+    /// The input replaces arguments with this token in the command.
+    #[serde(rename = "arg")]
+    Arg(String),
```

Comment:
I'm not sure about this change/implementation from the user-facing perspective and the schema - it moves the schema from a set of enum values (in JSON Schema syntax) to a `oneOf` that is more complicated and harder for users to adjust/grok.

I would propose an enum value like `ReplaceArg` (`replaceArg` in the JSON) that _always_ replaces a static string argument in the list, instead of allowing an arbitrary string.

We could have that argument be something like `{{instance_json}}` to be very unlikely to conflict with any actual arguments.

Then the manifest would look like:

```json
"set": {
  "executable": "foo",
  "args": ["config", "set", "--input", "{{instance_json}}"],
  "input": "replaceArg"
}
```

From a schema perspective, we can then validate that when `input` is `replaceArg` that `args` includes the static string and not when `input` is any other value.

Are there cases where users will need to specify a custom value for the replacement string?

```yaml
allOf:
  - if: { properties: { input: { const: replaceArg } } }
    then:
      properties:
        args:
          contains:  { const: '{{instance_json}}' }
          minContains: 1
          maxContains: 1
    else:
      properties:
        args:
          not: { contains:  { const: '{{instance_json}}' } }
```

Reply:
@michaeltlombardi please open a new issue to discuss